### DISCUSSION TITLE
[Feature] Change / Transfer Owner permissions of Media

### DISCUSSION THREAD

**Discussion by mirisbowring** [immich-app/immich#7885] [Feature Request]
### The feature

Hi,

i started using immich more than a year ago. Initially, i moved most of our photos into my account to have them stored somewhere.
During the past months, more users have been onboarded onto my immich instance (family members, friends, etc.).
Now i have the case, that i have many photos that do not exactly belong to me (e.g. i digitized a photo dvd from my sisters birthday party when she was like 5). In my example, i still have good memories regarding this event, but those images should belong to her.

For now, if i detect images in my timeline that do belong to another user, i download them and upload them via cli into their account. This is not a good process.
Also some of my friends stumbled across the same thing.

Therofre my feature request is:

"I would like to be able to select media / albums and press an [change owner] button and select one of the existing users so that the images are beeing transferred to their library."

There are some questions that should be considered beforehand:

- Should the image just appear in the others user timeline without beeing noticed or should the user accept the transfer befor? Probably he don't even want those photos or another user is trolling by pushing many cats into another users timeline.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by flesser** [comment]
I would really love to see such a feature, too.

Possibly a duplicate of #1992?

**Comment by hexxone** [comment]
I would also love to see this feature for Albums!
I dont think it makes a lot of sense to do this on a per-picture basis.
The Album-sharing feature already exists and has the permission/sharing gui already.

This way, when a new Owner gets promoted for an Album, it would simply "switch around" so the previous owner becomes an "Editor" and now has the album under his "shared" ones.
Then when transferring an Album, it could just update the Description to say "Album was transferred from User XYZ ... [original_text]".

Sure, there could be some trolling due to this, but one could maybe have a per-user permission setting checkbox "Allow transferring Album Ownership". So it could be enabled when needed for certain users, or in general only work for administrators..
Also I think you should generally have some faith in the other users ^^ Or their access should be reconsidered.

**Reply by mblennegard** [reply]
I definitely think it should be handled on a photo level, which is also where the main problem lies in my opinion.

Currently, it is possible to make another user as Editor for an album as you mention, but the fact is that the underlying photos will still be read-only for everyone except the Owner (= the user who uploaded the photos in the first place). This prevents e.g. people/faces to be added to shared photos.

However, defining permissions on the album level should contain an option whether or not to change the permission on all underlying photos as well or only for the album.

**Comment by flemmingss** [comment]
I support this FR.
In my use case I need to change the owner of the Album, and let all elements (pictures and videos) inside the album stay owned by however owns them.

A solution that should work for everyone:
- Make it possible to change owner for both Album and element.

**Comment by Craftefix** [comment]
I need this feature, I have some albums I nedd to transfer to other people

**Comment by pvonkaenel** [comment]
I only started using Immich yesterday, so I'm still exploring import and sharing options. I think I've found a workable approach for sharing a large family photo archive with my wife and adult kids using a dedicated archive account for the historical import.
The approach: the archive account partners with my wife and me (giving us full timeline access and the ability to create our own albums), while the kids get access via a shared album. What's nice about this is that partnership doesn't need to be bidirectional — the archive account shares outward without cluttering its own timeline.

This works reasonably well, but there are two gaps that ownership transfer would solve:

1. **Explore limitations** — Faces and Places in the Explore panel are restricted to assets you own, so my wife and I don't get that benefit from the archive account's photos even as partners.
2. **Album granularity** — I can create albums from a partner's photos, but I can't truly "own" a curated subset of the archive for finer personal organization.

Ideally I'd love two related features:

- **Ownership transfer** — move assets from one account to another permanently
- **Clone to my account** — create an independent copy of an asset under my own account, preserving all original metadata, while optionally keeping the original in the archive account

Either would significantly improve the multi-user family sharing experience.

**Comment by romatou** [comment]
+1 to this feature

**Comment by CritLoren** [comment]
Yes please, I just wanted to use this now and it's odd that it's not present, is it not just a database switch?

**Reply by flemmingss** [reply]
I hope with this change it will be easyer to implement:

_Road to 
v3.0.0 ....
album.owner has been relocated to album.users (with a role of owner)_

https://immich.app/blog/2026-april-recap?utm_source=chatgpt.com

**Reply by csuno** [reply]
> I hope with this change it will be easyer to implement:
> 
> _Road to v3.0.0 .... album.owner has been relocated to album.users (with a role of owner)_
> 
> https://immich.app/blog/2026-april-recap?utm_source=chatgpt.com

we already have 3.X and its not implemented ;(

**Comment by scottvpb** [comment]
I am also here for this feature.   The use is I have aging family members.  I know in the near future that they will not be able to manage their photos and I want to have albums converted into something that a group of family members  can curate, edit, tag people and update descriptions to aid in preservation.

**Comment by Astromormy** [comment]
Another +1 for this feature. Literally just setup OAuth sign-in and now want to reassign ownership of all my old account's media to my new one. Shame this hasn't been added after over two years.

**Comment by psilves1** [comment]
It would be great for multiple users to add/remove images from an album.

Ex: having a shared kiosk where you can update the album with new photos over time

**Comment by csuno** [comment]
Why is this "closed"!?

Our Familymembers regulary creates new Albums. Thats fine, but we cant share it with other by link. So we have to ask them to reshare it in an Album owned by us. We dont want to have the ownership of the pictures it self. Just the posibility to take the Albumownership!

And an administrativ option to disallow "Albumcreation" for some users.
